Tools/src/Tools/HgcrocEmulator.cxx

```diff
@@ -163,30 +173,40 @@ bool HgcrocEmulator::digitize(
         if (toa == 0) toa = 1;
         if (toa > 1023) toa = 1023;
       }
-
-      digiToAdd.emplace_back(
-          false, true,  // mark as a TOT measurement
-          (iADC > 0) ? digiToAdd.at(iADC - 1).adc_t()
-                     : pedestal,  // ADC t-1 is first measurement
-          tdc_counts,             // TOT
-          toa                     // TOA is third measurement
-      );
+      if (verbose)
+        ldmx_log(debug) << "    Adding TOT hit with toa = " << toa
+                        << ", tdc_counts = " << tdc_counts
+                        << " adc_t at prev iADC = "
+                        << digiToAdd.at(iADC - 1).adc_t();
+      // ADC at t-1
+      auto adc_at_tminus1 =
+          (iADC > 0) ? digiToAdd.at(iADC - 1).adc_t() : pedestal;
+      auto i_tot_sample = digiToAdd.size() - 1;
```

Comment:
```suggestion
      auto i_tot_sample = digiToAdd.size();
```
we haven't `emplace_back` the TOT sample yet so the index of the TOT sample is the current size of the vector. (If this line is moved to after the `emplace_back` directly below on line 185, then you would keep the `-1` in).